src/azure-cli-core/azure/cli/core/parser.py

```diff
@@ -150,18 +150,25 @@ def _check_value(self, action, value):
         # Override to customize the error message when a argument is not among the available choices
         # converted value must be one of the choices (if specified)
         if action.choices is not None and value not in action.choices:
-            error_msg = "{prog}: '{value}' is not an {prog} command. See '{prog} --help'.".format(prog=self.prog,
-                                                                                                  value=value)
+            if not self.command_source:
+                # parser has no `command_source`, value is part of command itself
+                error_msg = "{prog}: '{value}' is not in the '{prog}' command group. See '{prog} --help'.".format(
+                    prog=self.prog, value=value)
+            else:
+                # `command_source` indicates command values have been parsed, value is an argument
+                parameter = action.option_strings[0] if action.option_strings else action.dest
+                error_msg = "{prog}: '{value}' is not a valid value for '{param}'. See '{prog} --help'.".format(
+                    prog=self.prog, value=value, param=parameter)
             telemetry.set_user_fault(error_msg)
             logger.error(error_msg)
-            candidates = difflib.get_close_matches(value, action.choices, cutoff=0.8)
+            candidates = difflib.get_close_matches(value, action.choices, cutoff=0.7)
```

Comment:
What's the rationale behind this change?

Reply:
I felt it a little too strict; `0.8` allows zero edits for words under 5 letters; and at most 1 edit until words of length 10 or more. 
`0.7` allows a edit for four-character words and 2 edits for words of length 7, 8, and 9.

This isn't the main purpose of this PR, however, and I can change this back if you guys prefer

Reply:
I'm fine with it--just wanted to know the reason.